### ISSUE TITLE
deno_registry2 must reject broken module

### ISSUE THREAD

**Issue by elycheikhsmail** [denoland/deno_registry2#239]
I think deno registry must check  the module befor add them to deno.land/x store, if the code is invalid (like deno deploy check)  or have'nt description inform the contributor. this will be help develop to get "save" code from deno.land/x, this may need some convention about enty/entries poins for modules or manifest.

**Comment by wperron** [comment]
We already have process that goes through every file to `deno cache` them to analyze the dependencies (which is then reflected on the build status returned by the webhook) -- is that what you had in mind?

**Comment by elycheikhsmail** [comment]
> We already have process that goes through every file to `deno cache` them to analyze the dependencies (which is then reflected on the build status returned by the webhook) -- is that what you had in mind?

I deploy test_manager, I clean (not intended) one file content commit this brok my code, then I publish can't be used by any body because is invalid I'd like  deno registry to reject invalid code and notify dev deploying this.
if my not clear a nought I will rewite it.

**Comment by wperron** [comment]
Yeah I'm sorry, I'm not 100% clear on what you're asking here? "Broken code" could mean a couple of different things, it could mean that imports and broken, it could mean that there's syntax errors and it doesn't parse in v8, or it could mean a variety of other things. What's the flow you're looking for? What kind of errors do you want the registry to catch and how?

**Comment by moncefplastin07** [comment]
> Yeah I'm sorry, I'm not 100% clear on what you're asking here? "Broken code" could mean a couple of different things, it could mean that imports and broken, it could mean that there's syntax errors and it doesn't parse in v8, or it could mean a variety of other things. What's the flow you're looking for? What kind of errors do you want the registry to catch and how?

I think he means that you should check the work of the third party units before publishing them on the Deno registry
I think the piece of code below explains what he means 😄
```javascript
if(checkIsValidThirParty(incomingThirdPartyModule)){
     publishToDenoRegistry(incomingThirdPartyModule)
}else{
     makeReportToModuleOwner(incomingThirdPartyModule)
}
```

**Comment by MierenManz** [comment]
> Yeah I'm sorry, I'm not 100% clear on what you're asking here? "Broken code" could mean a couple of different things, it could mean that imports and broken, it could mean that there's syntax errors and it doesn't parse in v8, or it could mean a variety of other things. What's the flow you're looking for? What kind of errors do you want the registry to catch and how?

I would assume that broken code means that it is code that won't run.
like 
```ts
functi on ree() {
  return "ree";
}
```
And if things like this happen then the registry should reject the code.
